### PR TITLE
Card translations

### LINKED ISSUE #508: Translations for Cards
**Is your feature request related to a problem? Please describe.**
Many people that use github-readme-stats speak other languages. There are translations of the docs for this reason, but no translations for the cards.

**Describe the solution you'd like**
A query parameter like `?lang=EN`.

**Describe alternatives you've considered**
Most people's READMEs that I've seen just show the cards in english.

**Additional context**
I'm already working on a PR. I only know english, so if you have translations, please comment on this issue.

### Full List of words needing to be Translated:
* Template
* Archived
* [name]'s GitHub Stats
* Total Stars
* Total Commits
* Total Issues
* Total PRs
* Contributed to
* Most Used Languages
* Wakatime Week Stats
* No coding activity this week

### PR DESCRIPTION
**Pull request by nthnchu**
Add translations for the cards. Fixes #508.

- [x] Add translation functionality (with Google Translate)
- [x] Add tests and documentation
- [ ] Get translations verified or improved.
  - [ ] Chinese
  - [x] German
  - [ ] Spanish
  - [ ] French
  - [x] Itallian
  - [ ] Japanese
  - [ ] Korean
  - [ ] Portuguese
- [x] Maybe move translations to one file (like `src/common/translations.js`) instead of in each `src/cards/*` file.

### CODE REVIEW

**Comment by lrusso96** [comment]
@anuraghazra You should merge this to a new branch, otherwise we cannot commit on it!
Alternatively, @nthnchu you should give the writing permission to your fork, so I can apply the changes before the PR to master branch.

**Comment by nthnchu** [comment]
@lrusso96 You can fork my repo's card-translations branch and make a PR to that if I am correct.

**Comment by lrusso96** [comment]
> 
> 
> @lrusso96 You can fork my repo's card-translations branch and make a PR to that if I am correct.

Yep, you're right @nthnchu. I have opened a PR right now.

File: src/cards/stats-card.js [hunk 1]
```diff
@@ -79,37 +80,107 @@ const renderStatsCard = (stats = {}, options = { hide: [] }) => {
...
+      es: `Estadísticas de GitHub de ${encodeHTML(name)}`,
+      fr: `Statistiques GitHub de ${encodeHTML(name)}`,
+      it: `Statistiche GitHub di ${encodeHTML(name)}`,
+      ja: `${encodeHTML(name)}のGitHub統計`,
+      kr: `${encodeHTML(name)}의 GitHub 통계`,
+      "pt-br": `Estatísticas do GitHub de ${encodeHTML(name)}`,
+    },
+    stars: {
+      cn: "总星数",
+      de: "Insgesamt Sterne",
```
Comment (schmelto) [inline src/cards/stats-card.js:98]: Sterne insgesamt

File: src/cards/stats-card.js [hunk 2]
```diff
@@ -79,37 +80,107 @@ const renderStatsCard = (stats = {}, options = { hide: [] }) => {
...
+      es: "Estrellas totales",
+      fr: "Total d'étoiles",
+      it: "Stelle totali",
+      ja: "星の合計",
+      kr: "총 별",
+      "pt-br": "Total de estrelas",
+    },
+    commits: {
+      cn: "总承诺",
+      de: "Total Commits",
```
Comment (schmelto) [inline src/cards/stats-card.js:109]: Anzahl Commits

File: src/cards/stats-card.js [hunk 3]
```diff
@@ -79,37 +80,107 @@ const renderStatsCard = (stats = {}, options = { hide: [] }) => {
...
+      es: "RP totales",
+      fr: "Total des PR",
+      it: "PR totali",
+      ja: "合計PR",
+      kr: "총 PR",
+      "pt-br": "Total de PRs",
+    },
+    issues: {
+      cn: "总发行量",
+      de: "Gesamtausgaben",
```
Comment (schmelto) [inline src/cards/stats-card.js:131]: Anzahl Issues

File: src/cards/wakatime-card.js [hunk 4]
```diff
@@ -60,8 +60,34 @@ const renderWakatimeCard = (stats = {}, options = { hide: [] }) => {
     theme = "default",
     hide_progress,
     custom_title,
+    lang = "en",
   } = options;
 
+  const translations = {
+    title: {
+      cn: "Wakatime周统计",
+      de: "Wakatime Week Stats",
```
Comment (schmelto) [inline src/cards/wakatime-card.js:69]: Wakatime Wochen Status

**Review by schmelto — CHANGES_REQUESTED** [review]

File: api/index.js [hunk 5]
```diff
@@ -28,6 +28,7 @@ module.exports = async (req, res) => {
     theme,
     cache_seconds,
     custom_title,
+    lang,
```
Comment (anuraghazra) [inline api/index.js:31]: I think `locale` would be a better name for the prop since lang would be confusing since we have top-langs cards too.

**Comment by anuraghazra** [comment]
Hi @nthnchu thanks for the awesome work.

I've further improved the code by adding a I18n.js class to handle the internationalization, & also extracted all the translations to translation.js file for better colocation.

Feel free to improve on it & give feedbacks if you have any.

**Comment by nthnchu** [comment]
@anuraghazra I just changed it so you don't have to reference `translations.repocard["repocard.archived"]` and instead can do `translations.repocard.archived` (with the real variable names, I forget what they are called)

File: src/translations.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    es: "Estadísticas de la semana de Wakatime",
+    fr: "Statistiques de la semaine Wakatime",
+    it: "Statistiche della settimana di Wakatime",
+    ja: "ワカタイムウィーク統計",
+    kr: "Wakatime 주간 통계",
+    "pt-br": "Estatísticas da semana Wakatime",
+  },
+  "wakatimecard.nocodingactivity": {
+    cn: "本周没有编码活动",
+    de: "Keine Codierungsaktivität diese Woche",
```
Comment (schmelto) [inline src/translations.js:127]: de: "Keine Aktivitäten in dieser Woche ",

**Review by schmelto — CHANGES_REQUESTED** [review]

**Comment by anuraghazra** [comment]
> @anuraghazra I just changed it so you don't have to reference `translations.repocard["repocard.archived"]` and instead can do `translations.repocard.archived` (with the real variable names, I forget what they are called)

Actually it was totally intentional.

So that we have some context of which translation we are fetching just by looking at the `i18n.t("repocard.title")`  code.

**Comment by nthnchu** [comment]
> > @anuraghazra I just changed it so you don't have to reference `translations.repocard["repocard.archived"]` and instead can do `translations.repocard.archived` (with the real variable names, I forget what they are called)
> 
> Actually it was totally intentional.
> 
> So that we have some context of which translation we are fetching just by looking at the `i18n.t("repocard.title")` code.

Ah ok that makes sense. I'll revert my commit

**Comment by anuraghazra** [comment]
The translation code looks good but there is a few design issues with the card, because of the new languages, the text width is increased and the text is overlapping with the stats.

**Comment by anuraghazra** [comment]
Okay @nthnchu i think this is ready, now people can improve the translations on separate PRs

**Comment by anuraghazra** [comment]
Thanks for this feature @nthnchu & all of you who helped translating & will be going to help with the translation.

**Pull request merged**